src/librustc_mir/interpret/memory.rs

```diff
@@ -416,11 +416,11 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
         id: AllocId,
     ) -> InterpResult<'tcx, &mut Allocation<M::PointerTag, M::AllocExtra>> {
         let tcx = self.tcx;
-        let memory_extra = &self.extra;
+        let alloc = Self::get_static_alloc(id, tcx, &self);
```

Comment:
Oh, I had entirely missed this! This is not what we want. This is the reason for the perf regression. We don't want to ask for a static alloc if we don't absolutely need it. You turned the vast majority of the `get_mut` calls from one to *two* `HashMap` lookups.

Reply:
This could also explain why `MIRI_BACKTRACE=1` made everything so awfully slow when I used it for debugging on CI.

Reply:
Oh I remember that change, that was because there was no way to pass `self`  into the `get_mut_or` without having a mutable and an immutable reference at the same time.

Reply:
Yeah, that's why I am reverting to just `MemoryExtra` now.

Next time you have to reorder code like that in a way that moves an operation out of a closure and makes it executed much more often than previously... please call that out when submitting the PR, so we can review that specifically. :)  I should have seen this in the review, but it gets easier when you call out the "most interesting" bits.

Reply:
Yeah sorry, that was a little careless from my part. Next time will do